components/nexus-mime/src/test/java/org/sonatype/nexus/mime/internal/DefaultMimeSupportTest.java

```diff
@@ -87,35 +75,33 @@ public void testGuessfakeMimeRulesSourceMimeTypeFromPath() {
     MimeRulesSource source = new MimeRulesSource()
     {
       @Override
-      public String getRuleForPath(String path) {
-        return "foo/bar";
+      public MimeRule getRuleForName(String path) {
+        return new MimeRule(false, "foo/bar");
       }
     };
 
-    assertThat(underTest.guessMimeTypeFromPath(source, "/some/path/artifact.pom"), equalTo("foo/bar"));
-    assertThat(underTest.guessMimeTypeFromPath(source, "/some/path/artifact.jar"), equalTo("foo/bar"));
-    assertThat(underTest.guessMimeTypeFromPath(source, "/some/path/artifact-sources.jar"), equalTo("foo/bar"));
-    assertThat(underTest.guessMimeTypeFromPath(source, "/some/path/maven-metadata.xml"), equalTo("foo/bar"));
-    assertThat(underTest.guessMimeTypeFromPath(source, "/some/path/some.xml"), equalTo("foo/bar"));
-    assertThat(underTest.guessMimeTypeFromPath(source, "/some/path/some.tar.gz"), equalTo("foo/bar"));
-    assertThat(underTest.guessMimeTypeFromPath(source, "/some/path/some.tar.bz2"), equalTo("foo/bar"));
-    assertThat(underTest.guessMimeTypeFromPath(source, "/some/path/some.zip"), equalTo("foo/bar"));
-    assertThat(underTest.guessMimeTypeFromPath(source, "/some/path/some.war"), equalTo("foo/bar"));
-    assertThat(underTest.guessMimeTypeFromPath(source, "/some/path/some.ear"), equalTo("foo/bar"));
-    assertThat(underTest.guessMimeTypeFromPath(source, "/some/path/some.ejb"), equalTo("foo/bar"));
+    assertThat(underTest.guessMimeTypeFromPath("/some/path/artifact.pom", source), equalTo("foo/bar"));
+    assertThat(underTest.guessMimeTypeFromPath("/some/path/artifact.jar", source), equalTo("foo/bar"));
+    assertThat(underTest.guessMimeTypeFromPath("/some/path/artifact-sources.jar", source), equalTo("foo/bar"));
+    assertThat(underTest.guessMimeTypeFromPath("/some/path/maven-metadata.xml", source), equalTo("foo/bar"));
+    assertThat(underTest.guessMimeTypeFromPath("/some/path/some.xml", source), equalTo("foo/bar"));
+    assertThat(underTest.guessMimeTypeFromPath("/some/path/some.tar.gz", source), equalTo("foo/bar"));
+    assertThat(underTest.guessMimeTypeFromPath("/some/path/some.tar.bz2", source), equalTo("foo/bar"));
+    assertThat(underTest.guessMimeTypeFromPath("/some/path/some.zip", source), equalTo("foo/bar"));
+    assertThat(underTest.guessMimeTypeFromPath("/some/path/some.war", source), equalTo("foo/bar"));
+    assertThat(underTest.guessMimeTypeFromPath("/some/path/some.ear", source), equalTo("foo/bar"));
+    assertThat(underTest.guessMimeTypeFromPath("/some/path/some.ejb", source), equalTo("foo/bar"));
   }
 
   @Test
   public void testGuessNullMimeRulesSourceMimeTypeFromPath() {
```

Comment:
Test name reflects old API, "null" means "when no MimeRuleSource" is provided (as it is nullable). Before the commit 9981cefbccf2ee281cf2207e353df7a0e4d1b143 the API was "doubled" for calls with and without source, and by Stuart's recommendations it was changed to use varargs to make it more concise and simpler. Will rename the test.